### ISSUE TITLE
Is it possibe to proxy also Audio/Video Streams ?

### ISSUE THREAD

**Issue by ProTechEx** [morph1904/TygerCaddy#126]
Hi there,

i would use the TyperCaddy Reverse Proxy Server to proxy the Audio/Video Streams.
We have installed it successful. We have the following issues:

We have add a test Proxy Server: http://prntscr.com/nzlfay
And also add following Host: http://prntscr.com/nzlfky

Our test domain: http://proxytv.xxxxxxx.tv:25461 does successfully redirects to our streaming main server:   http://

and it does also does open the Playlist from there. But the Playlist does contain the IP Adress as streaming links like:
http://xxx.xxx.xxx.xxx:25461/proxyuser/proxypass/15755 
instead of the Domain:
http://proxytv.xxxxxxx.tv:25461/proxyuser/proxypass/15755 

If I redirect the Domain directly to the Main Server IP (without using the Proxy) and download the Playlist, so, in this case, the Playlist contains the correct streaming link, like:
http://proxytv.xxxxxxx.tv:25461/proxyuser/proxypass/15755

Very important is also,
If i try to open the direct stream url, so i can play the stream via VLC.
If the link is direct by the proxy, so VLC can't open the stream.

So basically, are there some special or custom configurations on the proxy settings to redirect also video streams?

Our target is to protect the IP of our Mainserver.

Thanks and regards.

**Comment by arevindh** [comment]
Try changing these options while editing the proxy. ![image](https://user-images.githubusercontent.com/693151/59162923-ef30b980-8b16-11e9-8bd5-964075a1be3b.png)

**Comment by morph1904** [comment]
The actual proxy server that is proxying the request is called Caddy.. https://caddyserver.com, The docs there are great, be aware though not all of Caddy's features are implemented in the GUI of TygerCaddy. Tyger2 will hope to address this. 

arevindh is correct with the websocket and the transparent options though. This will tell Caddy to pass the request through whilst making it appear as though there is no proxy server in use. 

I hope this helps. Let me know if you need any further support.

**Comment by morph1904** [comment]
Hi I assume you have this sorted. I will close this for now. Let me know if you have any further problems.

**Comment by ProTechEx** [comment]
> 
> 
> Try changing these options while editing the proxy.
> ![image](https://user-images.githubusercontent.com/693151/59162923-ef30b980-8b16-11e9-8bd5-964075a1be3b.png)

The Option "Transparent" solve the issue with getting the correct epoxying streaming links like:
http://proxytv.xxxxxxx.tv:25461/proxyuser/proxypass/15755

But the streams still not open even though the source Link is correct.

Just for basic info, if I open the Link on a Browser without using the proxy, I get a download file with type: video/mp2t   ( http://prntscr.com/o0anbe )

I just read also the caddyserver wiki and community, i have there something with the MIME Types and Header Settings, but i am very new and i don't know about those settings.

I just see here: http://prntscr.com/o0b1cc some configuration options, but i don't know about their possible values and settings.

Maybe someone can help me out, to set the correct values for the type: video/mp2t 

Thank you guys in front up for your support here.

**Comment by arevindh** [comment]
Tygercaddy is just a GUI for [Caddy](https://caddyserver.com/).  
You might find better luck in finding answers for your question on their [forums](https://caddy.community/). 
Please include what server you use for " Audio/Video Streams." and/or their protocols when asking in the forum.

**Comment by ProTechEx** [comment]
After my request ( https://caddy.community/t/media-video-mp2t-reverse-proxying-possible/5835 ) on their Forum, they write:

There’s no special configuration required. Caddy’s reverse proxy is designed to faithfully reproduce the upstream response for the connecting client, including any headers such as Content-Type.

**Comment by arevindh** [comment]
Which server are you using for serving media streams?

**Comment by ProTechEx** [comment]
Xtream Codes Panel (it based on ffmpeg and nginx)

iOS için Outlook<https://aka.ms/o0ukef> uygulamasını edinin
________________________________
Gönderen: itsmesid <notifications@github.com>
Gönderildi: Thursday, June 13, 2019 5:15:15 PM
Kime: morph1904/TygerCaddy
Bilgi: ProTechEx; Author
Konu: Re: [morph1904/TygerCaddy] Is it possibe to proxy also Audio/Video Streams ? (#126)


Which server are you using for serving media streams?

**Comment by arevindh** [comment]
Where can I get a sample one to try it out?

**Comment by ProTechEx** [comment]
Sure I need the IP of the Server, so I can redirect the Domain to the Proxy IP

iOS için Outlook<https://aka.ms/o0ukef> uygulamasını edinin
________________________________
Gönderen: itsmesid <notifications@github.com>
Gönderildi: Thursday, June 13, 2019 5:18:00 PM
Kime: morph1904/TygerCaddy
Bilgi: ProTechEx; Author
Konu: Re: [morph1904/TygerCaddy] Is it possibe to proxy also Audio/Video Streams ? (#126)


Where can I get a sample one to try it out?

**Comment by arevindh** [comment]
Can we chat here https://gitter.im/TygerCaddyUI/community

**Comment by ProTechEx** [comment]
I am a way, it would be better if can add me on Skype: pro-tech-ex

iOS için Outlook<https://aka.ms/o0ukef> uygulamasını edinin
________________________________
Gönderen: itsmesid <notifications@github.com>
Gönderildi: Thursday, June 13, 2019 5:22:54 PM
Kime: morph1904/TygerCaddy
Bilgi: ProTechEx; Author
Konu: Re: [morph1904/TygerCaddy] Is it possibe to proxy also Audio/Video Streams ? (#126)


Can we chat here https://gitter.im/TygerCaddyUI/community

**Comment by arevindh** [comment]
Seems Skype is not working for me, can you join the link

**Comment by ProTechEx** [comment]
I am joined in, but nobody answer me

iOS için Outlook<https://aka.ms/o0ukef> uygulamasını edinin
________________________________
Gönderen: itsmesid <notifications@github.com>
Gönderildi: Thursday, June 13, 2019 5:31:38 PM
Kime: morph1904/TygerCaddy
Bilgi: ProTechEx; Author
Konu: Re: [morph1904/TygerCaddy] Is it possibe to proxy also Audio/Video Streams ? (#126)


Seems Skype is not working for me, can you join the link

**Comment by arevindh** [comment]
The file you are trying to proxy is an M3U file.

> An M3U file is a plain text file that specifies the locations of one or more media files. 
> https://en.wikipedia.org/wiki/M3U

So the location you need to proxy is the final location provided in the file ( which redirects to some other location once opened ) 

TLDR -  you can only proxy through caddy, you need something else to parse the final URL and set it to caddy which is a fairly tedious task.
